modules/javafx.graphics/src/main/java/com/sun/glass/ui/headless/HeadlessApplication.java

```diff
@@ -0,0 +1,218 @@
+        setEventThread(eventThread);
+        eventThread.start();
+    }
+
+    @Override
+    protected void _invokeAndWait(Runnable runnable) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void _invokeLater(Runnable runnable) {
+        processor.invokeLater(runnable);
+    }
+
+    @Override
+    protected Object _enterNestedEventLoop() {
+        return processor.newRunLoop();
+    }
+
+    @Override
+    protected void _leaveNestedEventLoop(Object retValue) {
+        processor.leaveCurrentLoop(retValue);
+    }
+
+    @Override
+    protected int _isKeyLocked(int keyCode) {
+        return KeyEvent.KEY_LOCK_OFF;
+    }
+
+    @Override
+    public Window createWindow(Window owner, Screen screen, int styleMask) {
+        HeadlessWindow window = new HeadlessWindow(windowManager, owner, screen, frameBuffer, styleMask);
+        if (this.activeRobot != null) {
+            activeRobot.windowAdded(window);
+            window.setRobot(this.activeRobot);
+        }
+        return window;
+    }
+
+    @Override
+    public View createView() {
+        return new HeadlessView();
+    }
+
+    @Override
+    public Cursor createCursor(int type) {
+        this.cursor = new HeadlessCursor(type);
+        return this.cursor;
+    }
+
+    @Override
+    public Cursor createCursor(int x, int y, Pixels pixels) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected void staticCursor_setVisible(boolean visible) {
+        throw new UnsupportedOperationException("Not supported yet.");
+    }
+
+    @Override
+    protected Size staticCursor_getBestSize(int width, int height) {
+        return new Size(16, 16);
+    }
+
+    @Override
+    public Pixels createPixels(int width, int height, ByteBuffer data) {
+        return new HeadlessPixels(width, height, data);
+    }
+
+    @Override
+    public Pixels createPixels(int width, int height, ByteBuffer data, float scalex, float scaley) {
+        return new HeadlessPixels(width, height, data, scalex, scaley);
+    }
+
+    @Override
+    public Pixels createPixels(int width, int height, IntBuffer data) {
+        return new HeadlessPixels(width, height, data);
+    }
+
+    @Override
+    public Pixels createPixels(int width, int height, IntBuffer data, float scalex, float scaley) {
+        return new HeadlessPixels(width, height, data, scalex, scaley);
+    }
+
+    @Override
+    protected int staticPixels_getNativeFormat() {
+        return Pixels.Format.BYTE_BGRA_PRE;
+    }
+
+    @Override
+    public GlassRobot createRobot() {
+        this.activeRobot = new HeadlessRobot(this);
+        return this.activeRobot;
+    }
+
+    @Override
+    protected double staticScreen_getVideoRefreshPeriod() {
+        return 0.;
+    }
+
+    @Override
+    protected Screen[] staticScreen_getScreens() {
+        final int screenWidth = 1000;
```

Comment:
Should this be something that can be configured by the developer? E.g. a simple system property with a default value of 1000?

Reply:
This is a good question, and I believe it is part of a broader discussion. There are a number of places (e.g. here) where I use hardcoded, default values which should be fine for most usecases I'm aware of. However, I believe there is definitely value in having more configurable values (e.g. in screen dimensions, but also number of screens, default window sizes, multiclick values etc).
The challenge with this is that we need some way to have developers manipulate these values, for example using system properties (as you state as well). This is then adding a feature that doesn't exist yet, so it would require a deeper discussion on how to make the Headless platform configurable. I'm absolutely +1 on that, but I believe it would be easier if we do that as a follow-up?

Reply:
> I'm absolutely +1 on that, but I believe it would be easier if we do that as a follow-up?

I agree, makes sense. This way, we can also gradually add properties as (if) needed.